### PR TITLE
Add hdenabled flag in getwalletinfo

### PR DESCRIPTION
**Pull request by jeromer**
When running `bitcoin-cli getwalletinfo` we can see a new flag called `hdenabled` which is either `true` or `false`. This replicates the behaviour we see in `bitcoin-qt` which displays a "HD Enabled" logo in `BitcoinGUI::setHDStatus()`

The help now look like this:

    bitcoin-cli getwalletinfo help
    error code: -1
    error message:
    getwalletinfo
    Returns an object containing various wallet state info.

    Result:
    {
      "walletname": xxxxx,             (string) the wallet name
      "walletversion": xxxxx,          (numeric) the wallet version
      "balance": xxxxxxx,              (numeric) the total confirmed balance of the wallet in BTC
      "unconfirmed_balance": xxx,      (numeric) the total unconfirmed balance of the wallet in BTC
      "immature_balance": xxxxxx,      (numeric) the total immature balance of the wallet in BTC
      "txcount": xxxxxxx,              (numeric) the total number of transactions in the wallet
      "keypoololdest": xxxxxx,         (numeric) the timestamp (seconds since Unix epoch) of the oldest pre-generated key in the key pool
      "keypoolsize": xxxx,             (numeric) how many new keys are pre-generated (only counts external keys)
      "keypoolsize_hd_internal": xxxx, (numeric) how many new keys are pre-generated for internal use (used for change outputs, only appears if the wallet is using this feature, otherwise external keys are used)
      "unlocked_until": ttt,           (numeric) the timestamp in seconds since epoch (midnight Jan 1 1970 GMT) that the wallet is unlocked for transfers, or 0 if the wallet is locked
      "paytxfee": x.xxxx,              (numeric) the transaction fee configuration, set in BTC/kB
      "hdmasterkeyid": "<hash160>",    (string) the Hash160 of the HD master pubkey
      "hdenabled": true|false          (boolean) true if HD is enabled, false otherwise
    }

    Examples:
    > bitcoin-cli getwalletinfo
    > curl --user myusername --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "getwalletinfo", "params": [] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/

And the command output:

    bitcoin-cli getwalletinfo
    {
      "walletname": "wallet.dat",
      "walletversion": 139900,
      "balance": 0.00999772,
      "unconfirmed_balance": 0.00000000,
      "immature_balance": 0.00000000,
      "txcount": 7,
      "keypoololdest": 1514388183,
      "keypoolsize": 1000,
      "keypoolsize_hd_internal": 1000,
      "paytxfee": 0.00000000,
      "hdmasterkeyid": "ad304ceaa7bf2bf0a3543526968ee958fb94c847",
      "hdenabled": true
    }

### CODE REVIEW

**Comment by jonasschnelli** [comment]
It's kinda redundant. If `hdmasterkeyid` is present, hd is enabled. Maybe better to mention this in the RPC help instead of adding the redundant `hdenabled`?

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -2704,7 +2704,8 @@ UniValue getwalletinfo(const JSONRPCRequest& request)
             "  \"keypoolsize_hd_internal\": xxxx, (numeric) how many new keys are pre-generated for internal use (used for change outputs, only appears if the wallet is using this feature, otherwise external keys are used)\n"
             "  \"unlocked_until\": ttt,           (numeric) the timestamp in seconds since epoch (midnight Jan 1 1970 GMT) that the wallet is unlocked for transfers, or 0 if the wallet is locked\n"
             "  \"paytxfee\": x.xxxx,              (numeric) the transaction fee configuration, set in " + CURRENCY_UNIT + "/kB\n"
-            "  \"hdmasterkeyid\": \"<hash160>\"     (string) the Hash160 of the HD master pubkey\n"
+            "  \"hdmasterkeyid\": \"<hash160>\",    (string) the Hash160 of the HD master pubkey\n"
```
Comment (promag) [inline src/wallet/rpcwallet.cpp]: Agree with @jonasschnelli, just improve this description like `(only present when HD is enabled)`.

**Review by promag — COMMENTED** [review]
NACK, avoid redundant fields.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (promag) [reply src/wallet/rpcwallet.cpp]: Also add `optional` flag.

**Comment by jeromer** [comment]
Updating the RPC docs is a good option indeed.

**Comment by MeshCollider** [comment]
@jeromer concept ACK on updating the help text, thanks for contributing :)

**Comment by MarcoFalke** [comment]
Don't forget to squash your commits when you push the fixed-up version. See https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

**Comment by promag** [comment]
@jeromer do you plan to change the PR?

**Comment by gmaxwell** [comment]
The really important piece of information-- does the wallet contain any non-HD private keys-- appears to exist nowhere.

**Comment by jonasschnelli** [comment]
@gmaxwell point is not directly related to this PR, but an important one.

**Comment by fanquake** [comment]
I've opened #12293 to finish this up.

**Pull request closed without merging**